scripts/us_epa/superfund/sites/tar_creek/process_report2020.py

```diff
@@ -0,0 +1,208 @@
+                    inplace=True)
+            df.rename(columns={'Su lfate': 'Sulfate'}, inplace=True)
+        if idx == 1:
+            df = df[~df['Unnamed: 0'].str.contains('Averages')]
+            df['Unnamed: 0'] = df['Unnamed: 0'].replace(
+                {
+                    "/ ": "/",
+                    " /": "/",
+                    r'/ [^0-9]+': r'/[^0-9]+',
+                    'To tals': 'Totals',
+                    'To tal s': 'Totals',
+                    'Total s': 'Totals',
+                    'Tota ls': 'Totals',
+                    'To ta ls': 'Totals',
+                    'Disso lved': 'Dissolved',
+                    'Di ssolved': 'Dissolved',
+                    'Di sso lved': 'Dissolved',
+                    '200A': '2004',
+                    '19.E': '19.8'
+                },
+                regex=True)
+            df['Unnamed: 0'] = df['Unnamed: 0'].replace(
+                {'Dissolved': 'NaN Dissolved'})
+            df[['observationDate',
+                'contaminantType']] = df['Unnamed: 0'].str.split(n=1,
+                                                                 expand=True)
+            df['observationDate'] = pd.to_datetime(
+                df['observationDate']).ffill()
+            df[['temp', 'pH']] = df['Temp. pH'].str.split(n=1, expand=True)
+            df.drop(columns=['Unnamed: 0', 'Unnamed: 1', 'Temp. pH'],
+                    inplace=True)
+            df.rename(columns={"0 .0 .": "D.O."}, inplace=True)
+        if idx == 0:
+            df['Unnamed: 0'] = df['Unnamed: 0'].replace(
+                {
+                    "/ ": "/",
+                    " /": "/",
+                    r'/ [^0-9]+': r'/[^0-9]+',
+                    'To tals': 'Totals',
+                    'To tal s': 'Totals',
+                    'Total s': 'Totals',
+                    'Tota ls': 'Totals',
+                    'Disso lved': 'Dissolved',
+                    'Di ssolved': 'Dissolved',
+                    'Di sso lved': 'Dissolved',
+                    '200A': '2004',
+                    '19.E': '19.8'
+                },
+                regex=True)
+            df['Cond.'] = df['Cond.'].replace({'4U': '412'}, regex=True)
+            df['Sulfate'] = df['Sulfate'].replace({'U2': '122'}, regex=True)
+            df['Iron'] = df['Iron'].replace({'O.lU': '0.112'}, regex=True)
+            df['Unnamed: 0'] = df['Unnamed: 0'].replace(
+                {'Dissolved': 'NaN Dissolved'})
+            df[['observationDate',
+                'contaminantType']] = df['Unnamed: 0'].str.split(n=1,
+                                                                 expand=True)
+            df['observationDate'] = pd.to_datetime(
+                df['observationDate']).ffill()
+            df[['temp', 'pH']] = df['Temp. pH'].str.split(n=1, expand=True)
+            df['pH'] = df['pH'].replace({'.8 7.81': ''}, regex=True)
+
+            df.drop(columns=['Unnamed: 0', 'Unnamed: 1', 'Temp. pH'],
```

Comment:
why are we dropping these columns after fixing them in line 153, 172

Reply:
The following are the columns we drop while processing, since they were split into single columns:
- `Unnamed: 0` - Date of the observation and contaminant Type (Total or Dissolved) occur in the same column, separated by a space.
-  `Unnamed:1` -  an empty column from the extraction
-  `Temp. pH/ Temo. DH` - temperature and pH observations occur in the same column separated by a space
-  `Hardness Cadmium` - hardness of the water measured and the concentration of Cadmium occur in the same column and are space-separated.

Reply:
but `Unnamed: 0` is in the replace map .. should we just remove it from there?

Reply:
We first replace or fix the extracted values from `tabula`. 

Here, we refer to `Unnamed: 0` in the replace map to fix the values and then drop the `Unnamed: 0`  column from the dataframe. Hence, we will need to keep `Unnamed: 0` in the replace map.